lib/server.go

```diff
@@ -1981,17 +2009,13 @@ func (srv *Server) _addNewTxn(pp *Peer, txn *MsgDeSoTxn, rateLimit bool) ([]*Msg
 	chainState := srv.blockchain.chainState()
 	srv.blockchain.ChainLock.RUnlock()
 
-	if chainState != SyncStateFullyCurrent {
-		// We allow txn relay if chain is in a need blocks state and is running PoS.
-		// We will error in two cases:
-		// - the chainState is not need blocks state
-		// - the chainState is need blocks state but the chain is not on PoS.
-		if chainState != SyncStateNeedBlocksss || !srv.blockchain.params.IsPoSBlockHeight(tipHeight) {
-			err := fmt.Errorf("Server._addNewTxnAndRelay: Cannot process txn "+
-				"from peer %v while syncing: %v %v", pp, srv.blockchain.chainState(), txn.Hash())
-			glog.Error(err)
-			return nil, err
-		}
+	if chainState != SyncStateFullyCurrent && !srv.blockchain.params.IsPoSBlockHeight(tipHeight) {
```

Comment:
Did we change this to fix the SYNCING_HEADERS error on Diamond? I think it's fine to keep regardless just wondering.

Reply:
Yes - this was the fix to allow us to allow transaction relay regardless of sync status.